src/main/java/pl/ctrlpkw/service/SuspiciousSelector.java

```diff
@@ -0,0 +1,100 @@
+package pl.ctrlpkw.service;
+
+import com.google.common.collect.Lists;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import pl.ctrlpkw.api.dto.BallotResult;
+import pl.ctrlpkw.model.read.QuorumConfiguration;
+import pl.ctrlpkw.model.read.QuorumConfigurationRepository;
+import pl.ctrlpkw.model.write.Protocol;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+public class SuspiciousSelector implements ResultsSelectorStrategy {
+
+    private QuorumConfigurationRepository quorumConfigurationRepository;
+    private List<QuorumConfiguration> configurationEntries;
+
+    @Autowired
+    public SuspiciousSelector(QuorumConfigurationRepository quorumConfigurationRepository) {
+        this.quorumConfigurationRepository = quorumConfigurationRepository;
+        this.configurationEntries = Lists.newArrayList(quorumConfigurationRepository.allOrderByFromSizeDesc());
+    }
+
+    @Override
+    public Optional<BallotResult> apply(List<Protocol> wardProtocols) {
+        wardProtocols = getNonDepreciated(wardProtocols);
+
+        if (wardProtocols.size() == 0 ) return Optional.empty();
```

Comment:
A czy ten warunek jest w ogóle potrzebny? Co zwróci to mapowanie dla pustej listy?